corehq/apps/userreports/expressions/specs.py

```diff
@@ -400,3 +400,18 @@ def __call__(self, item, context=None):
             return string_value.split(self.delimiter)[index_value]
         except IndexError:
             return None
+
+
+class CoalesceExpressionSpec(JsonObject):
+    type = TypeProperty('coalesce')
+    expression = DictProperty(required=True)
+    default_expression = DictProperty(required=True)
+
+    def configure(self, expression, default_expression):
+        self._expression = expression
+        self._default_expression = default_expression
+
+    def __call__(self, item, context=None):
+        expression_value = self._expression(item, context)
+        default_value = self._default_expression(item, context)
+        return expression_value or default_value
```

Comment:
is the intention that you want this to use `default_value` on falsey things like `0`? i could see that behavior being a bit surprising.